src/test/java/net/openhft/chronicle/queue/TestDeleteQueueFile.java

```diff
@@ -189,6 +191,49 @@ public void firstAndLastIndexAreRefreshedAfterForceRefreshInterval() throws IOEx
         }
     }
 
+    @Test
+    public void tailingThroughDeletedCyclesWillRefreshThenRetry_Writable() throws IOException {
+        tailingThroughDeletedCyclesWillRefreshThenRetry(qwcd -> qwcd.queue);
+    }
+
+    @Test
+    public void tailingThroughDeletedCyclesWillRefreshThenRetry_ReadOnly() throws IOException {
+        tailingThroughDeletedCyclesWillRefreshThenRetry(qwcd -> SingleChronicleQueueBuilder.binary(qwcd.queue.fileAbsolutePath())
+                .rollCycle(RollCycles.FAST_DAILY)
+                .readOnly(true)
+                .build());
+    }
```

Comment:
This manifested in different ways for read-only and writable queues, I'm not entirely sure why, but I think it was to do with the fact they use different `DirectoryListing` implementations